stock-tracker/src/features/keyStats/component/keyStats.tsx

```diff
@@ -3,13 +3,14 @@ import { useSelector } from "react-redux";
 import Loading from "../../loading/component/loading";
 import ErrorMessage from "../../error/error";
 import "./KeyStats.css";
+import { AppState } from "../../../store/rootReducer";
 
 export const KeyStats = () => {
-  const response = useSelector(state => state.keyStats.response);
-  const loading = useSelector(state => state.keyStats.loading);
-  const error = useSelector(state => state.keyStats.error);
+  const { response, loading, error } = useSelector(
+    (state: AppState) => state.keyStats
+  );
 
-  function numberWithCommas(x) {
+  function numberWithCommas(x: number) {
```

Comment:
This looks like a pure function, so I'd move it to above this `KeyStats` function (so it isn't re-created on every render per component.)

Reply:
Good idea thanks